template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveKnoxConfigProvider.java

```diff
@@ -23,17 +25,22 @@ public class HiveKnoxConfigProvider implements CmTemplateComponentConfigProvider
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject templatePreparationObject) {
         String filePerEvent = ConfigUtils.getSafetyValveProperty("hive.hook.proto.file.per.event", "true");
-
+        List<ApiClusterTemplateConfig> serviceConfigs = new ArrayList<>();
         String cdhVersion = templateProcessor.getVersion().orElse("");
         if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERAMANAGER_VERSION_7_1_0)) {
-            return List.of(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, filePerEvent));
+            serviceConfigs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, filePerEvent));
         } else {
             KerberosConfig kerberosConfigOpt = templatePreparationObject.getKerberosConfig().get();
             String realm = kerberosConfigOpt.getRealm();
             String principal = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.principal", "HTTP/_HOST@" + realm);
             String keytab = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.keytab", "hive.keytab");
-            return List.of(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, principal + keytab + filePerEvent));
+            serviceConfigs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, principal + keytab + filePerEvent));
+        }
+        if (CMRepositoryVersionUtil.isS3SslChannelModeSupported(cdhVersion, templatePreparationObject.getCloudPlatform())) {
```

Comment:
what about with azure?

Reply:
Not applicable for azure since it's for s3a which is for AWS and GCP only.